### PR TITLE
Configure ContentParts and ContentFields with IOptions<ContentPartOptions> or IOptions<ContentFieldOptions>

### PR DESCRIPTION
**Pull request by deanmarcussen**
Fixes https://github.com/OrchardCMS/OrchardCore/issues/1061

@sebastienros couple of options I've found so far here.

Not sure of the direction you want to take?
configuring ContentPartOptions is easy
but all the Handlers and Drivers etc, are Scoped services, so unless you had other plans for them, they still need to be registered in the DI container.

So I added a `ContentPartBuilder` and an initial extension for adding display drivers.

- Refer MarkdownPart code for example using the builder, 
- Refer AutoroutePart, just for simply doing it with `.Configure<>`()

Guess you'll want do something for Fields as well!

### CODE REVIEW

**Comment by sebastienros** [comment]
Right, I think we can postpone the drivers change. I was just thinking about perf. 

The activators should be easier to manage as we are now registering a type. Later we'll also be able to add metadata to the declaration, in the form of a property value dictionary, and each module could extend it. Kind of the same as what stereotype is for content types, if we want to categorize fields and parts. For instance we could easily mention that a part is good for front end, or categories for flow, widgets, ... And each module could provide more metadata on any part. But it can be done later by changing the internal structure of this list of parts.

**Comment by deanmarcussen** [comment]
I think this is ready for another look @sebastienros 

It's only the activator's and the `ContentDefinitionManager` that injected `IEnumerable<ContentPart> or <ContentField>`

I've kept them backward compatible for the moment, but have done all the `ContentType` and `ContentField` with the new `Configure` options arrangement.

File: src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ServiceColletionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace OrchardCore.ContentManagement
+{
+    public static class ServiceColletionExtensions
+    {
```
Comment (deanmarcussen) [inline src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ServiceColletionExtensions.cs]: Colletion ?

File: src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ServiceColletionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        }
+
+        /// <summary>
+        /// Registers a content field type.
+        /// </summary>
+        public static IServiceCollection AddContentField<TContentField>(this IServiceCollection serviceCollection)
+            where TContentField : ContentField
+        {
+            return serviceCollection.Configure<ContentOptions>(o => o.AddField<TContentField>());
+        }
```
Comment (deanmarcussen) [inline src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ServiceColletionExtensions.cs]: cool, so we can add to this with an Action if / when there becomes metadata to configure

File: src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs [hunk 3]
```diff
@@ -235,7 +235,7 @@ public ActionResult ListFilterPOST(ListContentsViewModel model)
 
         [HttpPost, ActionName("List")]
         [FormValueRequired("submit.BulkAction")]
-        public async Task<ActionResult> ListPOST(ContentOptions options, IEnumerable<int> itemIds)
+        public async Task<ActionResult> ListPOST(ViewModels.ContentOptions options, IEnumerable<int> itemIds)
         {
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs]: I don't know what this is, but I assume you know what you're doing here :)

**Pull request merged**